spark-connector/build.gradle.kts

```diff
@@ -14,26 +15,35 @@ repositories {
 
 val scalaVersion: String = project.properties["scalaVersion"] as? String ?: extra["defaultScalaVersion"].toString()
 val sparkVersion: String = libs.versions.spark.get()
+val sparkMajorVersion: String = sparkVersion.substringBeforeLast(".")
 val icebergVersion: String = libs.versions.iceberg.get()
 val kyuubiVersion: String = libs.versions.kyuubi.get()
 val scalaJava8CompatVersion: String = libs.versions.scala.java.compat.get()
 
 dependencies {
-  implementation(project(":api"))
   implementation(project(":clients:client-java-runtime", configuration = "shadow"))
-  implementation(project(":common"))
-  implementation(libs.bundles.log4j)
   implementation(libs.guava)
```

Comment:
Do we use guava? Guava is a common package and need to shade.

Reply:
Another important thing. Which dependency should we shade?

Reply:
we only dependent on `guava` `iceberg-spark-runtime` `kyuubi-spark-connector-hive` for now, `guava` is shaded

Reply:
we should shade most jars to avoid dependency.

Reply:
> we only dependent on `guava` `iceberg-spark-runtime` `kyuubi-spark-connector-hive` for now, `guava` is shaded

How did we shade guava? Could you paste the configuration?

Reply:
```
tasks.shadowJar {
  isZip64 = true
  archiveFileName.set("${rootProject.name}-spark-connector-runtime-${sparkMajorVersion}_$scalaVersion-$version.jar")
  archiveClassifier.set("")

  relocate("com.google", "com.datastrato.gravitino.shaded.com.google")
}
```

Reply:
I got it.